llvm/lib/Transforms/Instrumentation/AddressSanitizer.cpp

```diff
@@ -1149,7 +1152,7 @@ AddressSanitizerPass::AddressSanitizerPass(
     AsanCtorKind ConstructorKind)
     : Options(Options), UseGlobalGC(UseGlobalGC),
       UseOdrIndicator(UseOdrIndicator), DestructorKind(DestructorKind),
-      ConstructorKind(ClConstructorKind) {}
+      ConstructorKind(ConstructorKind) {}
```

Comment:
should check if the `cl::opt` was set instead of defaulting to `invalid` (can't remember the exact method, `hasValue()`?)

Reply:
(Opt.getNumOccurrences() > 0) ? Opt : ConstructorKind;

Reply:
and we don't need to introduce Invalid